### ISSUE TITLE
mongomock needs to declare dependency on setuptools (or stop using it)

### ISSUE THREAD

**Issue by wimglenn** [mongomock/mongomock#850]
Mongomock uses [setuptools](https://github.com/mongomock/mongomock/blob/ca41f27161c50d83d5ef8c899623084dd6f09a1d/mongomock/__version__.py#L1) unconditionally. However, setuptools (which provides `pkg_resources`) is not stdlib, and it is not guaranteed to be there. So it needs to be specified as a dependency, otherwise users can get:

```
$ python3 -c 'import mongomock'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/.../.venv/lib/python3.9/site-packages/mongomock/__init__.py", line 79, in <module>
    from mongomock.__version__ import __version__
  File "/.../.venv/lib/python3.9/site-packages/mongomock/__version__.py", line 1, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

The current depenency tree is missing setuptools:

```
$ johnnydep mongomock==4.1.2
name              summary
----------------  -----------------------------------------------------------
mongomock==4.1.2  Fake pymongo stub for testing simple MongoDB-dependent code
├── packaging     Core utilities for Python packages
└── sentinels     Various objects to denote special meanings in python
```

The alternative would be to stop importing `pkg_resources`, which is deprecated anyway (https://github.com/mongomock/mongomock/issues/848), and use stdlib [`importlib.metadata`](https://docs.python.org/3/library/importlib.metadata.html) to retrieve the installed version instead.

**Comment by stur86** [comment]
I support switching to the newer `importlib` implementation, the deprecation warning messages are quite annoying.

**Comment by mdomke** [comment]
This should be fixed (at least) in the latest release `4.2.0.post1`
